frontend/src/components/auth/PrivateRoute.tsx

```diff
@@ -17,11 +16,23 @@ const PrivateRoute: React.FC<PrivateRouteProps> = ({
 }: PrivateRouteProps) => {
   const { authenticatedUser } = useContext(AuthContext);
 
-  return authenticatedUser ? (
+  if (authenticatedUser === null) {
+    return (
+      <Redirect to={LOGIN_PAGE} />
+    )
+  }
+
+  if (authenticatedUser.verified === false) {
+    if (!window.location.pathname.endsWith("/verification")) {
```

Comment:
Nit: React router has a built in way to do this

```
import { useLocation } from 'react-router-dom';
const location = useLocation();
const currentPath = location.pathname;
```

Also I'm assuming you're checking this because otherwise you'll be stuck on an infinite loop on the verification page?